logrlint.go

```diff
@@ -79,9 +79,15 @@ func decorateVendoredFuncs(entryFuncs stringSet, currentPkgImport, packageImport
 			if !strings.HasSuffix(importOrReceiver, ")") {
 				continue // invalid pattern
 			}
+			pointerChar := ""
 
-			leftOver := strings.TrimPrefix(importOrReceiver, "("+packageImport+".")
-			importOrReceiver = fmt.Sprintf("(%s.%s", currentPkgImport, leftOver)
+			if fn[1] == '*' {
+				pointerChar = "*"
+			}
+			prefix := "(" + pointerChar + packageImport + "."
+
+			leftOver := strings.TrimPrefix(importOrReceiver, prefix)
+			importOrReceiver = fmt.Sprintf("(%s%s.%s", pointerChar, currentPkgImport, leftOver)
```

Comment:
```go
			var pointerIndicator string
			if strings.HasPrefix(importOrReceiver[1:], "*") { // pointer type
				pointerIndicator = "*"
			}
			leftOver := strings.TrimPrefix(importOrReceiver, "("+packageImport+".")
			importOrReceiver = fmt.Sprintf("(%s%s.%s", pointerIndicator, currentPkgImport, leftOver)
```

Reply:
And your change to extract `prefix` alone is good. So it's just a bit rename

Reply:
done